halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/ShowCommand.java

```diff
@@ -18,8 +18,11 @@
 
 import com.beust.jcommander.Parameter;
 import com.beust.jcommander.Parameters;
+import com.netflix.spinnaker.halyard.cli.execute.v1.ExecuteShow;
```

Comment:
It's more than structural simplicity. It's encapsulating the appropriate details.

Giving the classes limited roles is a reasonable thing to do. And having an internal model that defines those classes and can provides an interpreter that mediates among them symbolically so that control flow is standardized and you're just plugging in specialized classes to implement the different activities is a common pattern. What's missing here is that usually that pattern treats those classes as a clique defined by what's being operated on, not the type of operation.

Rather than Executions {Insert, Delete }, Command {Insert, Delete}
You want  Insert {Execution, Command}  Delete {Execution, Command}

That gives encapsulation around the resource rather than the activity.

It is far more likely that you'd want cohesion around the different activities on a resource than around different resources for a given activity. The base classes are already defining the common parts of the activity anyway. Now you can add new resources without impacting other ones. Just drop a new module in. Otherwise you need to touch and modify all the other existing modules, not to say you've lose the encapsulation about the resource so are spreading that knowledge across all your modules as well.